serpentTools/xs.py

```diff
@@ -0,0 +1,481 @@
+            raise ValueError(
+                "Current number of perturbations is {}, not {}"
+                .format(len(self._perturbations), len(value)))
+        self._perturbations = value
+
+    @property
+    def states(self):
+        """
+        Iterable describing the names or values of each perturbation
+        branch. Length is equal to that of :attr:`perturbations`, and
+        the ``i``-th index of ``states`` indicates the values
+        perturbation ``perturbations[i]`` experiences.
+        """
+        return self._states
+
+    @states.setter
+    def states(self, value):
+        if self._states is None:
+            raise AttributeError(
+                "Collect first to ensure correct data structure.")
+        if len(value) != len(self._states):
+            raise ValueError(
+                "Current number of perturbations is {}, not {}"
+                .format(len(self._states), len(value)))
+
+        # check to make sure all perturbation vectors in the
+        # requested value are of the same length
+        for index, pertVec in enumerate(self._states):
+            if len(value[index]) != len(pertVec):
+                raise ValueError(
+                    "Current number of perturbations for state {} is {}, "
+                    "not {}"
+                    .format(self._perturbations[index], len(pertVec),
+                            len(value[index])))
+        self._states = value
+
+    @property
+    def axis(self):
+        """Tuple describing axis of underlying data"""
+        if self._axis is None:
+            raise AttributeError("Axis not set. Collect first.")
+        return self._axis
+
+    @axis.setter
+    def axis(self, value):
+        if self._axis is None:
+            raise AttributeError(
+                "Collect first to ensure correct data structure.")
+
+        # coerce into tuple to enforce some immutability
+        if not isinstance(value, tuple):
+            value = tuple(value)
+
+        if len(value) != len(self._axis):
+            raise ValueError(
+                "Current axis has {} dimensions, not {}"
+                .format(len(self._axis), len(value)))
+
+        self._axis = value
+
+    @property
+    def burnups(self):
+        """Vector of burnups from coefficient file"""
+        if self._burnups is None:
+            raise AttributeError("Burnups not set. Collect first.")
+        return self._burnups
+
+    @burnups.setter
+    def burnups(self, value):
+        if self._burnups is None:
+            raise AttributeError(
+                "Collect first to ensure correct data structure.")
+        if not isinstance(value, ndarray):
+            value = array(value)
+        if value.size != self._burnups.size:
+            raise ValueError(
+                "Current burnup vector has {} items, not {}"
+                .format(self._burnups.size, value.size))
+        self._burnups = value
+
+    def collect(self, perturbations, xsType):
```

Comment:
**NOTE** Only stores on set of group constants, e.g. just those that begin with `'inf'`. This leaves out some things like discontinuity factors currently. Can add at request for this PR or next